tests/plugins/test_koji_promote.py

```diff
@@ -1364,3 +1365,44 @@ def test_koji_promote_set_media_types(self, tmpdir, os_env, expect_result):
             assert sorted(image['media_types']) == sorted(expect_result)
         else:
             assert 'media_types' not in image.keys()
+
+    @pytest.mark.parametrize(('parent_output', 'expect_result'), [
+        (None, None),
+        ("1234", 1234),
+        ("string123", None),
+        (1234, 1234),
+        (1234890, 1234890),
+    ])
+    def test_koji_promote_set_parent_id(self, tmpdir, os_env, parent_output, expect_result):
+        session = MockedClientSession('')
+        tasker, workflow = mock_environment(tmpdir,
+                                            name='ns/name',
+                                            version='1.0',
+                                            release='1',
+                                            session=session)
+
+        if parent_output:
```

Comment:
Needs a test when `koji_parent` didn't run at all

Reply:
I am mocking what is in workflow, and runner is running only koji_promote

so how such a test has value?

Reply:
Having a use case where `PLUGIN_KOJI_PARENT_KEY` key is not set at all in `prebuild_results` would cover that.

Reply:
added